'use strict';
var util = require('util');
var HttpError = require('./../httpError');

module.exports = BadRequest;

/**
 * Error indicating an XSOData internal problem, e.g. used in try/catch statements to catch runtime errors
 * @param message
 * @param cause
 * @param context
 * @constructor
 */
function BadRequest(message, context, cause) {
    HttpError.call(this, message, context,cause, 400);
    this.context = context;
}

util.inherits(BadRequest, HttpError);

